index.js

```diff
@@ -1,31 +1,34 @@
-const ReadyResource = require('ready-resource')
+const { EventEmitter } = require('events')
 const Hypercore = require('hypercore')
 const b4a = require('b4a')
 const HypCrypto = require('hypercore-crypto')
 const IdEnc = require('hypercore-id-encoding')
+const safetyCatch = require('safety-catch')
 
-class PassiveCoreWatcher extends ReadyResource {
+class PassiveCoreWatcher extends EventEmitter {
   constructor (corestore, { watch, open }) {
     super()
-
     this.store = corestore
     this.watch = watch
     this.open = open
 
     this._oncoreopenBound = this._oncoreopen.bind(this)
     this._openCores = new Map()
-  }
 
-  async _open () {
-    this.store.watch(this._oncoreopenBound)
-    await Promise.all(
-      [...this.store.cores.map.values()].map(this._oncoreopenBound)
-    )
+    // Give time to add event handlers
+    setImmediate(() => {
```

Comment:
why is the immediate needed?

Reply:
Should have been queueMicroTask + needed a check on whether it was already destroyed. Pushed a fix.
The reason is to give time to add event-handlers before the background processing starts